Adding additinal tv channel card to the media player model

miloit opened the pull request:
Adding additinal tv channel card to the media player model

miloit committed:
Adding tvchannelmodel

miloit committed:
adding tv channel modell

miloit committed:
adding tv channel cards

miloit committed:
adding tv channel cards

adding tvchannel cards

miloit committed:
Merge branch 'master' of https://github.com/miloit/remote-software

martonborzak commented on components/media_player/ui/Card.qml:
```diff
@@ -195,13 +200,15 @@ Rectangle {
             id: cardSpeakerControl
-            Rectangle {
-                anchors.fill: parent
-                color: "green"
-            }
+            CardTVChannelList {}
```
This was reserved for the speaker control interface. Why remove it and add CardTVChannelList here as well? You've added it above.

miloit replied:
resolved

martonborzak commented on components/media_player/ui/CardTVChannelList:
```diff
@@ -0,0 +1,253 @@
+/******************************************************************************
+ *
+ * Copyright (C) 2018-2019 Marton Borzak <hello@martonborzak.com>
```
Please change the copyright to you. I didn't write this code :)

miloit replied:
resolved

zehnm replied:
It still says Marton ;)

martonborzak commented on components/media_player/ui/CardTVChannelList.qml:
```diff
@@ -0,0 +1,253 @@
+/******************************************************************************
+ *
+ * Copyright (C) 2018-2019 Marton Borzak <hello@martonborzak.com>
```
Please change the copyright to you.

miloit replied:
resolved

martonborzak commented on components/media_player/ui/ChannelView.qml:
```diff
@@ -0,0 +1,223 @@
+/******************************************************************************
+ *
+ * Copyright (C) 2018-2019 Marton Borzak <hello@martonborzak.com>
```
Same here.

miloit replied:
resolved

martonborzak commented on components/media_player/ui/TVChannelView.qml:
```diff
@@ -0,0 +1,223 @@
+/******************************************************************************
+ *
+ * Copyright (C) 2018-2019 Marton Borzak <hello@martonborzak.com>
```
Same here.

miloit replied:
resolved

martonborzak reviewed (changes requested):
Please exclude `.ts` files from the pull request.
The build fails, because the dependencies should be updated to use the new `integrations.library`.

All integrations needs to be updated if we merge this pull request.

miloit committed:
Update in code after review

removing ts files

removing ts files

removing ts files

removing ts files from commit

removing ts files from commit

miloit committed:
message

miloit committed:
deleted 1 file

miloit committed:
Delete .gitignore

miloit committed:
some minor changes for better overview

miloit committed:
Minor changes

miloit committed:
Adding a mediaplayer remote for sending commands to a media player

miloit committed:
Improvements

miloit committed:
code changes as requested

miloit committed:
Update to requested changes

miloit committed:
Deleted the definition out of the remote files as discussed
updated dependencies

miloit committed:
code style

martonborzak committed:
fixup: removed duplicate from schema

martonborzak committed:
fixup: update url format

martonborzak committed:
fixup: remove url format

martonborzak committed:
fixup: missing type in hardware-schema

martonborzak committed:
fixup: missing types

martonborzak committed:
fixup: hardware schema

miloit committed:
Fixing bottom menu for media player entity
Adding EPG View Card

miloit committed:
Adding Tooltip line for the remote

zehnm commented on dependencies.cfg:
```diff
@@ -3,4 +3,4 @@
 #   separator: colon + optional white space
 #   version: git branch or tag
-integrations.library: v0.5.1
+integrations.library: v0.5.3
```
Your integrations.library PR has been merged  and I've create a new release: `v0.6.0`
Please update the dependency, this will fix the build errors.

miloit committed:
Update dependencies.cfg

Update dependencies according integrations.library update

zehnm commented on components/media_player/ui/CardMediaPlayerEPGView.qml:
```diff
@@ -0,0 +1,519 @@
+                        mouseArea.onClicked: {
+                            Haptic.playEffect(Haptic.Click);
+                            contextMenuLoader.setSource("qrc:/basic_ui/ContextMenu.qml", { "width": itemFlickable.width, "id": item_key, "type": item_type, "list": item_commands })
+                        }
+                    }
+                }
+            }*/
+    /*}
+
+/* Item {
+
+            Loader {
+                id: channellistLoader
+                asynchronous: true
+                anchors.fill: parent
+            }
+
+            Text {
+                id: backButton
+                color: Style.color.text
+                text: Style.icon.left_arrow
+                renderType: Text.NativeRendering
+                width: 70; height: 70
+                verticalAlignment: Text.AlignVCenter; horizontalAlignment: Text.AlignHCenter
+                font {family: "icons"; pixelSize: 80 }
+                anchors { left: parent.left; leftMargin: 10; top: parent.top; topMargin: 20 }
+
+                MouseArea {
+                    id: backButtonMouseArea
+                    width: parent.width + 20; height: parent.height + 20
+                    anchors.centerIn: parent
+
+                    onClicked: {
+                        Haptic.playEffect(Haptic.Click);
+                        swipeView.currentIndex = 0;
+                    }
+                }
+            }
+        }*/
+    //}
```
Why is everything commented out?
Please remove file, implement functionality, or just show a "not implemented yet" message if it this view is required to be included.
If the functionality is not ready yet, submit an additional EPG pull request later.

zehnm commented on components/media_player/ui/CardMediaPlayerRemote.qml:
```diff
@@ -0,0 +1,540 @@
+        width: parent.width; height: parent.height
+        currentIndex: 0
+        interactive: false
+        clip: true*/
+
+    //Item {
+
+    /*Rectangle {
+            id: powerButton
+            visible: true
+            width: 118; height: 60
+            radius: height/2
+            color: Style.color.red
+            anchors { top: parent.top; left: parent.left }
+
+            Text {
+                color: Style.color.line
+                text: Style.icon.power_on
+                width: 70; height: 70
+                verticalAlignment: Text.AlignVCenter; horizontalAlignment: Text.AlignHCenter
+                font {family: "icons"; pixelSize: 60 }
+                anchors.centerIn: parent
+            }
+
+            MouseArea {
+                id: mouseArea
+                anchors.fill: parent
+                onClicked: {
+                    Haptic.playEffect(Haptic.Click);
+                    //if (obj.isSupported(Remote.F_POWER_TOGGLE)) {
+                    obj.play();
+                    /*} else if (obj.isSupported(Remote.F_POWER_ON) && obj.isSupported(Remote.F_POWER_OFF)) {
+                            if (obj.isOn)
+                                obj.powerOff();
+                            else
+                                obj.powerOn();
+                        }
+                }
+            }
+        }*/
```
As in `CardMediaPlayerEPGView.qml`: clean code ;-)

zehnm commented on components/media_player/ui/CardTVChannelList.qml:
```diff
@@ -0,0 +1,253 @@
+                        Text {
+                            id: title
+                            color: Style.color.text
+                            text: "TV Kanal"//qsTr("My playlists") + translateHandler.emptyString
```
Needs to be translated

zehnm commented on config-schema.json:
```diff
@@ -921,7 +932,6 @@
             },
             "updateUrl": {
               "type": "string",
-              "format": "url",
```
@martonborzak why did you remove the format specifier?
The format specifier makes sure, that the string conforms to an URL and that the user can't enter an invalid format (of course, if the tool or validator supports it).

An issue might be that `url` is not supported by all json tools. Well, that's the major issue with json IMHO, it has more proposals and different views than real specifications :-(
However, ajv validator supports it: https://github.com/ajv-validator/ajv/pull/402

An alternative would be:

```json
{
    "type": "string",
    "format": "uri",
    "pattern": "^(https?)://"
}
```

Note to myself: check the used json validator in remote-software, it it can handle `url` format.

martonborzak replied:
@zehnm the format was causing the json validation to fail.

zehnm replied:
Which validation? Runtime, GitHub action, ajv command line, web-configurator, code editor?

martonborzak replied:
The GitHub action. I haven't checked the others.

miloit commented:
Closed due to erronous deleting the repository
